### PR TITLE
feat(core/wordle): wordle

### PR DESCRIPTION
**Pull request by Seidko**
# preview
image

### CODE REVIEW

**Comment by MaikoTan** [comment]
I added a helper function in [koishijs/wordle@`f56cb45` (#1)](https://github.com/koishijs/wordle/pull/1/commits/f56cb45575539094e0174aa08bf9f78bfd2e0ea8) so then you could install `vscode-styled-components` extension to get a fully highlighted CSS.

**Comment by MaikoTan** [comment]
Also fixed linting

**Comment by Seidko** [comment]
<message forward>
<message>This is a XSS XD</message>
</message>

File: packages/wordle/src/index.ts [hunk 1]
```diff
@@ -1,13 +1,83 @@
...
+  }
 }
 
-export function apply(ctx: Context) {
-  // TODO: call wordle
+export namespace Wordle {
+  export const name = 'koishi-plugin-wordle'
+  export interface Config extends WordleCore.Config {}
+  export const Config: Schema<Config> = Schema.object({
+    imageMode: Schema.boolean().default(true),
```
Comment (MaikoTan) [inline packages/wordle/src/index.ts:79]: `imageMode` should be the configuration of the core, je pense.

File: packages/wordle/src/locales/zh-CN.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+commands:
+  wordle:
+    messages:
+      start: 开始你的wordle挑战！
+      win: 恭喜你获得胜利！
+      lose: 好可惜……再接再厉吧！
```
Comment (MaikoTan) [inline packages/wordle/src/locales/zh-CN.yml:6]: Just a lint complains, always append a new line at the end of file.

File: packages/core/src/locales/zh-CN.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+wordle:
+  core:
+    wordle: Wordle
+    text-mode: |
+      Wordle
```
Comment (MaikoTan) [inline packages/core/src/locales/zh-CN.yml:5]:
```suggestion
      {name}
```

Others games are not called `Wordle`, non hard-coding ici.

File: packages/core/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+  constructor(protected ctx: Context, protected config: WordleCore.Config) {
+    ctx.i18n.define('zh', require('./locales/zh-CN'))
+  }
+
+  abstract getTodayWord(): Promise<string>
+  abstract validateInput(input: string, solution: string): T[]
+  abstract getRandomWord(): string
+  public render(chars: T[][]): Element {
+    const textMode = !(this.config.imageMode && this.ctx.puppeteer)
```
Comment (MaikoTan) [inline packages/core/src/index.tsx:41]: Better to print a warning message says something like `WARN: No puppeteer installed, switch to text mode.`

**Review by MaikoTan — CHANGES_REQUESTED** [review]
Looks good to me, several comments but it is okay to merge.

File: packages/wordle/src/index.ts [hunk 5]
```diff
@@ -1,13 +1,83 @@
-import { WordleCore } from '@koishijs/wordle'
-import { Context } from 'koishi'
+import { WordleCore } from '@koishijs/wordle/src'
```
[the remark this answers is not in the document]
Comment (Seidko) [reply packages/wordle/src/index.ts:1]: but if I didn't add `src`, ts will start complain `cannot found module`.

[the remark this answers is not in the document]
Comment (MaikoTan) [reply packages/wordle/src/index.ts:1]: Mught there be something wrong about the tsconfig or IDEs, I would take a look later. But importing from `src` is incorrect way, please don't.

**Pull request closed without merging**